sklearn/metrics/_plot/cap_curve.py

```diff
@@ -0,0 +1,390 @@
+            The label of the positive class. When `pos_label=None`, if `y_true`
+            is in {-1, 1} or {0, 1}, `pos_label` is set to 1, otherwise an
+            error will be raised.
+
+        normalize_scale : bool, default=True
+            Whether to normalize values between 0 and 1 for the plot.
+
+        plot_chance_level : bool, default=True
+            Whether to plot the expected curve of a classifier whose
+            predictions are independent of the feature values passed to
+            `decision_function` or `predict_proba`.
+
+        name : str, default=None
+            Name of CAP curve for labeling. If `None`, name will be set to
+            `"Classifier"`.
+
+        ax : matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is created.
+
+        **kwargs : dict
+            Additional keywords arguments passed to matplotlib `plot` function.
+
+        Returns
+        -------
+        display : :class:`~sklearn.metrics.CAPCurveDisplay`
+            Object that stores computed values.
+        """
+        if pos_label is None:
+            pos_label = 1
+        if sample_weight is None:
+            sample_weight = np.ones_like(y_true, dtype=float)
+
+        pos_label_validated, name = cls._validate_from_predictions_params(
+            y_true, y_pred, sample_weight=sample_weight, pos_label=pos_label, name=name
+        )
+
+        # ensure y_true is boolean for positive class identification
+        y_bool = y_true == pos_label_validated
+
+        # sort predictions and true values based on the predictions
+        sorted_indices = np.argsort(y_pred)[::-1]
+        y_true_sorted = y_bool[sorted_indices]
+        sample_weight_sorted = sample_weight[sorted_indices]
+
+        # compute cumulative sums for true positives and all cases
+        y_true_cumulative = np.cumsum(y_true_sorted * sample_weight_sorted)
+        cumulative_total = np.cumsum(sample_weight_sorted)
+
+        viz = cls(
+            y_true_cumulative=y_true_cumulative,
+            cumulative_total=cumulative_total,
+            estimator_name=name,
+            pos_label=pos_label_validated,
+        )
+
+        return viz.plot(
+            ax=ax,
+            name=name,
+            normalize_scale=normalize_scale,
+            plot_chance_level=plot_chance_level,
+            **kwargs,
+        )
+
+    @classmethod
+    def from_estimator(
+        cls,
+        estimator,
+        X,
+        y,
+        *,
+        sample_weight=None,
+        response_method="auto",
+        pos_label=None,
+        normalize_scale=True,
+        plot_chance_level=True,
+        name=None,
+        ax=None,
+        **kwargs,
+    ):
+        """Create the Cumulative Accuracy Profile.
+
+        This is also known as a (cumulative) gain curve.
+
+        Parameters
+        ----------
+        estimator : BaseEstimator
+            A fitted estimator object implementing :term:`predict`,
```

Comment:
We should be explicit that this method only accepts binary classifier.

We will update the docstring later, in a follow-up PR that adds support for regressor.